aclocal.m4

```diff
@@ -162,7 +162,7 @@ AC_DEFUN([OCAML_AS_HAS_DEBUG_PREFIX_MAP], [
 
   # Modify C-compiler variables to use the assembler
   CC="$AS"
-  CFLAGS="--debug-prefix-map old=new -o conftest.$ac_objext"
+  CFLAGS="--debug-prefix-map=old=new -o conftest.$ac_objext"
```

Comment:
This is suspicious. The option when passed to gcc should be `-fdebug-prefix-map=old=new`, the binutils option (passed to `as`) is listed as `--debug-prefix-map old=new`.

Reply:
> The option when passed to gcc should be -fdebug-prefix-map=old=new, the binutils option (passed to as) is listed as --debug-prefix-map old=new.

Yes, we're not calling `as`, we're calling the C compiler (`$CC -c`) with an assembly file:
```
$ cat > test.S <<EOF
camlPervasives__loop_1128:
        .file   1       "pervasives.ml"
        .loc    1       193
EOF

# mingw's $cc doesn't like $as options
$ x86_64-w64-mingw32-gcc -c test.S -o test.o --debug-prefix-map old=new
x86_64-w64-mingw32-gcc: error: unrecognized command-line option ‘--debug-prefix-map’; did you mean ‘--debug-prefix-map=’?
$ x86_64-w64-mingw32-as -c test.S -o test.o --debug-prefix-map old=new

$ x86_64-w64-mingw32-gcc -c test.S -o test.o -fdebug-prefix-map old=new
x86_64-w64-mingw32-gcc: error: unrecognized command-line option ‘-fdebug-prefix-map’; did you mean ‘-fdebug-prefix-map=’?
$ x86_64-w64-mingw32-gcc -c test.S -o test.o -fdebug-prefix-map=old=new
```

Reply:
> Yes, but we're not calling as, we're calling the C compiler ($CC -c) with an assembly file

and I believe this has wider repercussions: although GCC let us (more or less) pass CLI option to the assembler through the C compiler, [clang does not #10770](https://github.com/ocaml/ocaml/issues/10770).